doc/source/data/transforming-datasets.rst

```diff
@@ -80,13 +80,18 @@ API in Datasets.
 
 A UDF can be a function or a callable class, which has the following input/output:
 
-* **Input type**: a ``pandas.DataFrame``, ``pyarrow.Table`` or a Python list. You can
-  control the input type fed to your UDF by specifying the ``batch_format`` parameter in
-  :meth:`.map_batches() <ray.data.Dataset.map_batches>`. By default, the ``batch_format``
-  is "native", which will feed ``pandas.DataFrame`` to UDF regardless of the underlying
-  batch type.
-* **Output type**: a ``pandas.DataFrame``, ``pyarrow.Table`` or a Python list. Note
-  the output type doesn't need to be the same as input type.
+* **Input type**: a ``pandas.DataFrame``, ``pyarrow.Table``, ``numpy.ndarray``,
+  column-dictionary of ``numpy.ndarray``, or a Python list. You can control the input
+  type fed to your UDF by specifying the ``batch_format`` parameter in
+  :meth:`.map_batches() <ray.data.Dataset.map_batches>`. By default, the
+  ``batch_format`` is ``"native"``, which will feed data batches in the
+  canonical/"native" format for each data type, namely: ``pandas.DataFrame``
+  batches on tabular (Arrow or Pandas) data, ``numpy.ndarray`` batches on tensor data,
+  and Python lists on simple Python list data.
+* **Output type**: a ``pandas.DataFrame``, ``pyarrow.Table``, ``numpy.ndarray``,
```

Comment:
This is an improvement, but I think the docs here could be a lot better. In particular:
- How about adding Tabbed examples, one for each input type?
- Tabbed examples for each output type?

Then, we can also get rid of / reduce in size the monolithic "UDF examples" code block.